### PR TITLE
feat: add metadata to transaction WebSocket messages

### PR DESCRIPTION
**Pull request by 0xApotheosis**
**Description**
Add metadata to transaction WebSocket messages to propagate the addition of metadata in https://github.com/shapeshift/unchained/pull/278.

### CODE REVIEW

File: packages/types/src/chain-adapters/index.ts [hunk 1]
```diff
@@ -137,6 +137,7 @@ export type SubscribeTxsMessage<T extends ChainTypes> = {
   status: TxStatus
   tradeDetails?: TradeDetails
   transfers: Array<TxTransfer>
+  data?: TxMetadata | undefined
```
Comment (kaladinlight) [inline packages/types/src/chain-adapters/index.ts:140]:
```suggestion
  data?: TxMetadata
```

File: packages/types/src/chain-adapters/index.ts [hunk 2]
```diff
@@ -150,6 +151,11 @@ export type TradeDetails = {
   type: TradeType
 }
 
+export interface TxMetadata {
+  method?: string
+  parser?: string
```
Comment (kaladinlight) [inline packages/types/src/chain-adapters/index.ts:156]: If parser metadata is included, should we at least require these two fields? I know this currently matches the payload coming from unchained, but just got me thinking.

**Review by kaladinlight — APPROVED** [review]
Two small comments, but looks good!

File: packages/types/src/chain-adapters/index.ts (hunk 2 above)
Comment (0xApotheosis) [reply packages/types/src/chain-adapters/index.ts:156]: That's a fair point, though at the moment `method` is always undefined for zrx parsing, as we haven't yet sourced and added an ABI for it.

File: packages/types/src/chain-adapters/index.ts (hunk 1 above)
Comment (0xApotheosis) [reply packages/types/src/chain-adapters/index.ts]: https://github.com/shapeshift/lib/pull/396/commits/501b22431f8b51ae1febae7c0635e324518a5381

File: packages/types/src/chain-adapters/index.ts (hunk 2 above)
Comment (kaladinlight) [reply packages/types/src/chain-adapters/index.ts:156]: True. Let's keep it in mind as a required base metadata type that we can extend if needed if it makes sense.

Comment (0xApotheosis) [reply packages/types/src/chain-adapters/index.ts]: Agree - I've made `parser` required for now in https://github.com/shapeshift/lib/pull/396/commits/b768e20811faf34b50379a38a189637b77e2336e.

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**